Sources/PixelExperimentKit/ExperimentEventTracker.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  ExperimentEventTracker.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public protocol ExperimentActionPixelStore {
+     func removeObject(forKey defaultName: String)
+     func integer(forKey defaultName: String) -> Int
+     func set(_ value: Int, forKey defaultName: String)
+ }
+
+public protocol ExperimentEventTracking {
+    /// Increments the count for a given event key and checks if the threshold has been exceeded.
+    ///
+    /// This method performs the following actions:
+    /// 1. If the `isInWindow` parameter is `false`, it removes the stored count for the key and returns `false`.
+    /// 2. If `isInWindow` is `true`, it increments the count for the key.
+    /// 3. If the updated count meets or exceeds the specified `threshold`, the stored count is removed, and the method returns `true`.
+    /// 4. If the updated count does not meet the threshold, it updates the count and returns `false`.
+    ///
+    /// - Parameters:
+    ///   - key: The key used to store and retrieve the count.
+    ///   - threshold: The count threshold that triggers a return of `true`.
+    ///   - isInWindow: A flag indicating if the count should be considered (e.g., within a time window).
+    /// - Returns: `true` if the threshold is exceeded and the count is reset, otherwise `false`.
+    func incrementAndCheckThreshold(forKey key: String, threshold: Int, isInWindow: Bool) -> Bool
```

Comment:
I think we could improve readability of this one by introducing typealias, e.g.
typealias ThresholdCheckResult = Bool
func updateCount(forKey key: String, threshold: Int, isInWindow: Bool) -> ThresholdCheckResult
WDYT?